### ISSUE TITLE
Improve smart/semantic selection: Capture URLs and wrap on TUI borders

### ISSUE THREAD

**Issue by shazow** [alacritty/alacritty#1210]
We already have a good basic implementation of #30 via #368 but there are a couple small issues with it:

## URLs aren't treated as a contiguous chunks

This is because of `:` is in the default set of `semantic_escape_chars`. I think it makes sense for a lot of scenarios (e.g. `filename:123`). Would it make sense to make an exception for this rule if the `:` is part of a _scheme_? That is, if it's followed immediately by `//`, such as `https://`.

## Wrapping from border characters doesn't work

The `semantic_escape_chars` list contains some common border characters like `│` and `|`. For example, when you have a tmux split.

Would it make sense to split out the `semantic_escape_chars` into an additional `semantic_border_chars` and treat the border characters as wrappable?

Bonus: we could check if the border character is extended below/above, to confirm that it's actually a border. This might not be very helpful though, eg if it's a 1 line box.

**Comment by chrisduerr** [comment]
I don't think this should be necessary, URLs will be easy to work with after #113 and semantic selection throughout the border characters would always put those in the selection, so if people desire to do so and go beyond it, they can always explicitly do so.

**Comment by shazow** [comment]
> I don't think this should be necessary, URLs will be easy to work with after #113

I have also been tracking and looking forward to #113, but click-to-open and smart-select are different use cases. A lot of the time I want to select a URL to copy it elsewhere, I explicitly don't want to open it. #113 does not help with this.

> semantic selection throughout the border characters would always put those in the selection, so if people desire to do so and go beyond it, they can always explicitly do so.

I don't understand what you mean here, could you rephrase/expand?

**Comment by nixpulvis** [comment]
I too would love to be able to double click a URL and have it be selected, and now that the task of figuring out what text contains URLs is being worked on in #1617 I feel like it might be the right time to tackle this. I'm aware of #1233, and taking a look at it now.

Just to be clear, I wouldn't want to break the ability to double click on paths.

**Comment by kchibisov** [comment]
With the custom regexes you could define an action just to select a url/your regex, like with `Alt + Click`.

We also support hyperlinks for a long time now.
